lib/lists.nix

```
@@ -1330,35 +1330,47 @@ rec {
           pairs);
 
   /**
-    Compare two lists element-by-element.
+    Compare two lists element-by-element with a comparison function `cmp`.
+
+    List elements are compared in order, the first non-equal pair of elements determines the result.
+
+    :::{.note}
+    The `<` operator can also be used to compare lists in a bi-state manner. (e.g. `[1 2] < [1 3]` is `true`).
```

```suggestion
    The `<` operator can also be used to compare lists using a boolean condition (e.g. `[1 2] < [1 3]` is `true`).
```

Bi-state tripped me up. It's not wrong but harder to read IMO.